lib/compress/zstd_compress.c

```diff
@@ -3370,6 +3371,14 @@ static size_t ZSTD_compressBlock_splitBlock_internal(ZSTD_CCtx* zc, void* dst, s
         cSize += cSizeChunk;
         currSeqStore = nextSeqStore;
     }
+
+    if (cSize > ZSTD_BLOCKSIZE_MAX + ZSTD_blockHeaderSize) {
```

Comment:
A few things I'm not sure to understand here : 
1) are you certain that, at this place in the code `cSize` includes the block header size ?
2) the limitation regarding the maximum compressed block size related to the complete block. Thus, in compressed format, it includes both the literals and sequence sub-blocks. Yet, below, I only notice `ZSTD_compressSequences`

Reply:
1. Yes, `ZSTD_compressSequences_singleBlock()` includes the block header
2. `ZSTD_compressSequences_singleBlock()`. This function compresses both the literals and the sequences (I had named it in the style of the existing `ZSTD_entropyCompressSequences()` which also compresses literals and sequences). Maybe a better name would be `ZSTD_compressSeqStore_singleBlock()` and rename the existing `ZSTD_entropyCompressSequences()` to `ZSTD_entropyCompressSeqStore()`?

Reply:
Yes, the new name is much better for clarity